Library/Homebrew/cmd/update-report.rb

```diff
@@ -183,6 +189,23 @@ def migrate_legacy_cache_if_necessary
     end
   end
 
+  def migrate_cache_entries_to_double_dashes(initial_version)
+    return if initial_version > "1.7.1"
+
+    HOMEBREW_CACHE.children.each do |child|
+      next unless child.file?
+
+      next unless /^(?<prefix>[^\.]+[^\-])\-(?<suffix>[^\-].*)/ =~ child.basename.to_s
+      target = HOMEBREW_CACHE/"#{prefix}--#{suffix}"
+
+      if target.exist?
+        FileUtils.rm_rf child
+      else
+        FileUtils.mv child, target, force: true
```

Comment:
I still reckon it's worth handling permission exceptions here a bit more nicely e.g. from below:

```ruby
     rescue Errno::EACCES, Errno::ENOENT
       ofail <<~EOS
         Could not create symlink at #{dst}!
         Please do this manually with:
           sudo ln -sf #{src} #{dst}
           sudo chown $(whoami) #{dst}
       EOS
     end
```

Reply:
Added warnings for permissions exceptions.